src/core/or/connection_edge.c

```diff
@@ -618,6 +617,11 @@ export_hs_client_circuit_id(const edge_connection_t *edge_conn,
     src_port = gid & 0x0000ffff;
   }
 
+  /* Grab the original dest port from the hs ident */
+  if (edge_conn->hs_ident) {
+    dst_port = edge_conn->hs_ident->orig_virtual_port;
+  }
```

Comment:
Can we really have a `dst_port = 0` that makes sense to send the HAProxy header?